### PR TITLE
Xml encoding fix

### LINKED ISSUE #1592: Some mails don't display (suspect xml mails)
Some messages will show up in the list of mail, and the tooltip mouserover preview will show that the content is xml (see attached image) 
mailpile-bugs-xml

however, clicking on the mail to open it fails, showing a "Hrm, We Could Not Find Anything" message on a largely blank screen.  No messages or errors are logged to the console, nor to the online event log.

### PR DESCRIPTION
**Pull request by JackDca**
Fixes #1592.

### CODE REVIEW

File: mailpile/mailutils.py [hunk 1]
```diff
@@ -1347,6 +1347,18 @@ def get_message_tree(self, want=None, tree=None):
 
                 if mimetype == 'text/html':
                     if want is None or 'html_parts' in want:
+                        # Find and delete possibly conflicting xml encoding
+                        # declaration to prevent lxml ValueError.
+                        # e.g. <?xml version="1.0" encoding="ISO-8859-1"?>
+                        re_encoding=re.compile('<\?xml version=[^ ?>]*'+
```
Comment (BjarniRunar) [inline mailpile/mailutils.py]: Can you add spaces around your = signs, to match the coding style elsewhere in the app?

Comment (BjarniRunar) [inline mailpile/mailutils.py]: (and other characters, not just = )

File: mailpile/mailutils.py [hunk 2]
```diff
@@ -1347,6 +1347,18 @@ def get_message_tree(self, want=None, tree=None):
...
+                                                '((?! +encoding=) [^ ?>]*)*'+
+                                                '( +encoding=[^ ?>]*)')
+                        enc_match=re_encoding.match(payload)
```
Comment (BjarniRunar) [inline mailpile/mailutils.py]: Could this be shortened a bit by using re.sub and substituting with the empty string?

Comment (JackDca) [inline mailpile/mailutils.py]: Re the above two comments - I will do some work on this - thanks for the suggestions.

**Comment by JackDca** [comment]
Changes requested by BjarniRunar have been pushed to xml-encoding-fix branch on JackDca/Mailpile.

Yes, that does shorten it a bit!

**Pull request merged**